### PR TITLE
add useQuery and useMutation

### PR DESCRIPTION
**Pull request by kwiedenhaupt**
fetch a level for a user in the database using the useQuery hook
update a user's level upon game completion using the useMutation hook
add a button to the winner game component where the user can save his/her progress
Currently these are done manually.

Need to: 
find out how to get the userId table object to match my variable in order to have it pull automatically
also add the same functionality for the fetchGameLevel function as well.

### CODE REVIEW

**Comment by kavanparam** [comment]
Nice work with the queries, mutations, and block click animations!

File: components/math/longestStreak/Winner.tsx [hunk 1]
```diff
@@ -1,36 +1,118 @@
...
+  const [resetGameLevel] = useMutation(RESET_GAME_LEVEL);
+
+  function handleResetGameLevel() {
+    resetGameLevel({
+      variables: {
+        userId: user.uid,
+      },
+      refetchQueries: [
+        {
+          query: RESET_GAME_LEVEL,
```
Comment (vithushan19) [inline components/math/longestStreak/Winner.tsx:34]: Why are you refetching this mutation here? I think only queries should be refetched

Comment (vithushan19) [reply components/math/longestStreak/Winner.tsx:34]: Unless this is supposed to be fetch game level?

File: graphql/longestStreak/fetchGameLevel.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    longestStreakUserData(where: { userId: { _eq: $userId } }) {
+      user {
+        name
+      }
+      currentLevel
+    }
+  }
+`;
+
+export type FetchGameLevelResponse = {
```
Comment (vithushan19) [inline graphql/longestStreak/fetchGameLevel.ts:14]: Nice

File: pages/api/longestStreak.ts [hunk 3]
```diff
@@ -82,39 +84,50 @@ export function longestSubarray(array: GameBlockState[], x: BlockState) {
...
-    BlockState.PLAYER_ONE_SELECTED && BlockState.HIGHLIGHTED
-  );
-  console.log("P1", playerOneArray);
+
+
+export function calculateWinner(array: GameBlockState[], functionName: () => void) {
+  let playerOneArray = longestSubarray(array, BlockState.PLAYER_ONE_SELECTED);
   let playerTwoArray = longestSubarray(array, BlockState.PLAYER_TWO_SELECTED);
-  console.log("P2", playerTwoArray);
+  let outcome = functionName()
```
Comment (vithushan19) [inline pages/api/longestStreak.ts:98]: This isn't clear what's happening. Can you rename things to be more descriptive or add comments

File: pages/api/longestStreak.ts [hunk 4]
```diff
@@ -82,39 +84,50 @@ export function longestSubarray(array: GameBlockState[], x: BlockState) {
...
-    array,
-    BlockState.PLAYER_ONE_SELECTED && BlockState.HIGHLIGHTED
-  );
-  return playerOneArray;
+
+
+
+export function showWinner() {
+  let optionOne = true;
+  let optionTwo = false;
```
Comment (vithushan19) [inline pages/api/longestStreak.ts:113]: What is going on here? Why is this needed

File: pages/api/showEndGameImage.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import Firework from "../../components/math/longestStreak/Firework";
+
+export function showEndGameImage() {
+  let optionOne = <Firework />;
```
Comment (vithushan19) [inline pages/api/showEndGameImage.tsx:4]: This should be a ts file not tsx

**Review by vithushan19 — APPROVED** [review]
Lgtm, feels like the code is starting to get messy and hard to read. I would go through your feature and look for ways to clean up the code

Looks like you got most of the functionality though.

File: package.json [hunk 6]
```diff
@@ -31,6 +31,7 @@
     "@sendgrid/mail": "^7.6.0",
     "@types/react-redux": "^7.1.16",
     "babel-jest": "^26.6.3",
+    "bootstrap": "^5.2.1",
```
[the remark this answers is not in the document]
Comment (kwiedenhaupt) [reply package.json:34]: It is not. I removed it.

File: components/math/longestStreak/Winner.tsx (hunk 1 above)
Comment (kwiedenhaupt) [reply components/math/longestStreak/Winner.tsx:34]: this i removed and put in the game.tsx

File: pages/api/longestStreak.ts (hunk 3 above)
Comment (kwiedenhaupt) [reply pages/api/longestStreak.ts:98]: comments added

File: pages/api/showEndGameImage.tsx (hunk 5 above)
Comment (kwiedenhaupt) [reply pages/api/showEndGameImage.tsx:4]: I could only get it to work in a tsx file due to the fact I'm conditionally rendering a component. Is there a better way?
